crates/forge/bin/cmd/test/mod.rs

```diff
@@ -364,9 +382,60 @@ impl TestArgs {
             "decode-internal",
             self.decode_internal.as_ref().and_then(|v| v.as_ref()),
         )?;
+        // maybe_override_mt("flamegraph", self.flamegraph.as_ref())?;
 
         let libraries = runner.libraries.clone();
-        let outcome = self.run_tests(runner, config, verbosity, &filter, &output).await?;
+        let mut outcome = self.run_tests(runner, config, verbosity, &filter, &output).await?;
+
+        if self.flamegraph || self.flamechart {
+            let (suite_name, (test_name, test_result)) = outcome
+                .results
+                .iter_mut()
+                .find(|(_, r)| !r.test_results.is_empty())
+                .map(|(suite_name, r)| (suite_name, r.test_results.iter_mut().next().unwrap()))
+                .unwrap();
+
+            let arena = test_result
+                .traces
+                .iter_mut()
+                .find_map(
+                    |(kind, arena)| {
+                        if *kind == TraceKind::Execution {
+                            Some(arena)
+                        } else {
+                            None
+                        }
+                    },
+                )
+                .unwrap();
+
+            // Decode traces.
+            let decoder = outcome.last_run_decoder.as_ref().unwrap();
+            decode_trace_arena(arena, decoder).await?;
+            let mut fst = folded_stack_trace::build(arena);
+
+            let label = if self.flamegraph { "flamegraph" } else { "flamechart" };
+            let contract = suite_name.split(':').last().unwrap();
+            let test_name = test_name.trim_end_matches("()");
+            let file_name =
+                format!("{}/{label}_{contract}_{test_name}.svg", std::env::temp_dir().display());
+            let file = std::fs::File::create(&file_name).wrap_err("failed to create file")?;
+
+            let mut options = inferno::flamegraph::Options::default();
+            options.title = format!("{label} {contract}::{test_name}");
+            options.count_name = "gas".to_string();
+            if self.flamechart {
+                options.flame_chart = true;
+                fst.reverse();
+            }
+
+            // Generate SVG.
+            inferno::flamegraph::from_lines(&mut options, fst.iter().map(|s| s.as_str()), file)
+                .wrap_err("failed to write svg")?;
+
+            // Open SVG in default program.
+            opener::open(&file_name).wrap_err("failed to open flamegraph")?;
```

Comment:
Let's handle the error gracefully here by saving flamegraph into `cache/{filename}` to allow user to still take a look at it even if `opener` failed to find a way to open it.

Reply:
I've pushed a commit to address this. Pls have a loook